great_expectations/rule_based_profiler/profiler.py

```diff
@@ -137,64 +159,58 @@ def __init__(
                 },
             )
 
-            parameter_builders: List[ParameterBuilder] = []
+            # Instantiate ParameterBuilders
+            parameter_builder_configs: List[ParameterBuilderConfig] = rule_config[
+                "parameter_builders"
+            ]
 
-            parameter_builder_configs: dict = rule_config.get("parameter_builders")
-
-            if parameter_builder_configs:
-                parameter_builder_config: dict
-                for parameter_builder_config in parameter_builder_configs:
-                    parameter_builders.append(
-                        instantiate_class_from_config(
-                            config=parameter_builder_config,
-                            runtime_environment={"data_context": data_context},
-                            config_defaults={
-                                "module_name": "great_expectations.rule_based_profiler.parameter_builder"
-                            },
-                        )
+            parameter_builders: List[ParameterBuilder] = []
+            for parameter_builder_config in parameter_builder_configs:
+                parameter_builders.append(
+                    instantiate_class_from_config(
+                        config=parameter_builder_config,
+                        runtime_environment={"data_context": data_context},
+                        config_defaults={
+                            "module_name": "great_expectations.rule_based_profiler.parameter_builder"
+                        },
                     )
+                )
```

Comment:
Is there a way to recursively run `instantiate_class_from_config`? I think this whole constructor could be simplified down to `instantiate_class_from_config(config=profiler_config, runtime_environment=..., config_defaults=...)`

Reply:
The issue with this would be passing down runtime_environment and config_defaults down the call stack (or knowing which values to pass down).

Reply:
Thanks for the pairing session on this @cdkini. As we saw in the `datasource` example, there isn't a simple way of running `instantiate_class_from_config()` recursively while also having the correct `config_defaults` passed in at every level. 

I think I'm ok with the loop here in the constructor, since it ensures that any error with the profiler instantiation happens but have a slight preference for splitting the code out into a helper method and having the constructor be something like `new_profiler = self.create_profiler_from_config(**rule_config)`